homeassistant/components/websocket_api/http.py

```diff
@@ -71,17 +73,29 @@ def __init__(self, hass: HomeAssistant, request: web.Request) -> None:
     async def _writer(self) -> None:
         """Write outgoing messages."""
         # Exceptions if Socket disconnected or cancelled by connection handler
+        to_write = self._to_write
+        logger = self._logger
+        wsock = self.wsock
         with suppress(RuntimeError, ConnectionResetError, *CANCELLATION_ERRORS):
             while not self.wsock.closed:
-                if (process := await self._to_write.get()) is None:
+                if (process := await to_write.get()) is None:
                     break
+                message = process if isinstance(process, str) else process()
 
-                if not isinstance(process, str):
-                    message: str = process()
-                else:
-                    message = process
-                self._logger.debug("Sending %s", message)
-                await self.wsock.send_str(message)
+                if not self.command_phase or to_write.empty():
+                    logger.debug("Sending %s", message)
+                    await wsock.send_str(message)
+                    continue
+
+                messages: list[str] = [message]
+                while not to_write.empty():
+                    if (process := to_write.get_nowait()) is None:
+                        break
```

Comment:
this is only breaking out of the while loop to get all the messages. Not the `while not self.wsock.closed:`

Reply:
loop in loop 🤦 

fixed